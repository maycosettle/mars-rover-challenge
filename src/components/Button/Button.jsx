import PropTypes from 'prop-types';

import useStyles from './styles';

const Button = ({ name, rounded, icon, color, bgColor, transparent, disabled, className, onClick }) => {
  const styles = useStyles({ rounded, color, bgColor, transparent, disabled, className });

  return (
    <button type="button" onClick={onClick} disabled={disabled} className={styles.button}>
      {icon && icon}
      {name}
    </button>
  );
};

Button.propTypes = {
  type: PropTypes.oneOf(['button', 'submit']),
  rounded: PropTypes.string,
  color: PropTypes.string,
  bgColor: PropTypes.string,
  transparent: PropTypes.bool,
  className: PropTypes.string,
  disabled: PropTypes.bool,
  onClick: PropTypes.func,
};

Button.defaultProps = {
  type: 'button',
  rounded: 'full',
  color: 'white',
  bgColor: 'primary',
  transparent: false,
  className: '',
  disabled: false,
  onClick: () => {},
};

export default Button;
