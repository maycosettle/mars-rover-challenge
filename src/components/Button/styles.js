import classNames from 'classnames';

const useStyles = ({ rounded, color, bgColor, transparent, disabled, className }) => {
  const container = classNames(
    `inline-flex items-center justify-center ${transparent ? '' : 'shadow-sm py-2 px-6 font-semibold'} select-none`,
    `rounded-${rounded}`
  );

  const border = classNames('border', {
    'border-transparent': bgColor !== 'white' || transparent,
    'border-gray-200': bgColor === 'white',
  });

  const text = classNames('text-base sm:text-sm', {
    [`text-${color}`]: !disabled,
    'text-gray-300': disabled,
  });

  const bg = classNames('transition-colors', {
    'focus:outline-none focus:ring-2 focus:ring-offset-2': !disabled,
    [`bg-${bgColor}-500 hover:bg-${bgColor}-700 focus:ring-${bgColor}-500`]:
      bgColor !== 'white' && !transparent && !disabled,
    [`bg-${bgColor}-50 hover:bg-${bgColor}-100 focus:ring-${bgColor}-100`]:
      bgColor !== 'white' && !transparent && !disabled,
    'bg-transparent': transparent,
    'bg-white hover:bg-gray-50': bgColor === 'white' && !disabled,
    [`${transparent ? 'bg-transparent' : 'bg-gray-100'} cursor-default pointer-events-none`]: disabled,
  });

  return { button: classNames(container, border, text, bg, className) };
};

export default useStyles;
