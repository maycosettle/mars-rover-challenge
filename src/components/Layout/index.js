export * from './PageLayout';
export * from './PaperLayout';
export * from './PhotoLayout';
