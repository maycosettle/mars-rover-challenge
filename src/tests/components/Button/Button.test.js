import { shallow } from 'enzyme';
import { Button } from 'components/Button';

describe('tests on <Button />', () => {
  const wrapper = shallow(<Button />);

  test('Should render Button correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
