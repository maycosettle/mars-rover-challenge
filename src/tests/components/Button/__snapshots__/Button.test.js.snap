// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on <Button /> Should render Button correctly 1`] = `
<button
  className="inline-flex items-center justify-center shadow-sm py-2 px-6 font-semibold select-none rounded-full border border-transparent text-base sm:text-sm text-white transition-colors focus:outline-none focus:ring-2 focus:ring-offset-2 bg-primary-500 hover:bg-primary-700 focus:ring-primary-500 bg-primary-50 hover:bg-primary-100 focus:ring-primary-100"
  disabled={false}
  onClick={[Function]}
  type="button"
/>
`;
